playbooks/roles/local.provision/templates/site_inventory.j2

```diff
@@ -0,0 +1,22 @@
+site:
+  hosts:
+{%- for host in groups['site'] +%}
+    {{ host }}:
+  {%- for var in ('host', 'port', 'user', 'ssh_private_key_file', 'ssh_common_args') +%}
+    {%- if hostvars[host]['ansible_' + var] is defined +%}
+      ansible_{{ var }}: {{ hostvars[host]['ansible_' + var] }}
+    {%- endif +%}
+  {%- endfor +%}
+      site:
+        {{ hostvars[host].site | to_nice_yaml(indent = 2) | indent(8) }}
+{%- endfor +%}
+
+{%- for group in groups +%}
+  {%- if group not in ('all', 'ungrouped', 'site') +%}
+{{ group }}:
+  hosts:
+    {%- for host in groups[group] +%}
+    {{ host }}:
```

Comment:
Do we really need the trailing colon(`:`)? Isn't this expected to be a list of hosts for a group?

Reply:
The yaml-based inventory format requires hosts to be objects (https://docs.ansible.com/ansible/latest/inventory_guide/intro_inventory.html#inventory-basics-formats-hosts-and-groups)

Reply:
Got it, thanks.